actors/miner/src/lib.rs

```diff
@@ -2003,6 +2002,225 @@ impl Actor {
+                sector_number: sector.sector_number,
+                seal_proof: params.seal_proof_type,
+                sealed_cid: sector.sealed_cid,
+                deprecated_deal_ids: vec![],
+                expiration: sector.expiration,
+                activation: curr_epoch,
+                deal_weight: DealWeight::zero(),
+                verified_deal_weight: DealWeight::zero(),
+                initial_pledge: sector_initial_pledge.clone(),
+                expected_day_reward: sector_day_reward.clone(),
+                expected_storage_pledge: sector_storage_pledge.clone(),
+                power_base_epoch: curr_epoch,
+                replaced_day_reward: TokenAmount::zero(),
+                sector_key_cid: None,
+                flags: SectorOnChainInfoFlags::SIMPLE_QA_POWER,
+            })
+            .collect::<Vec<SectorOnChainInfo>>();
+
+        let sectors_len = sectors_to_add.len();
+
+        let total_pledge = BigInt::from(sectors_len) * sector_initial_pledge;
+
+        let (needs_cron, fee_to_burn) = rt.transaction(|state: &mut State, rt| {
+            let current_balance = rt.current_balance();
+            let available_balance = state
+                .get_unlocked_balance(&current_balance)
+                .with_context_code(ExitCode::USR_ILLEGAL_STATE, || {
+                    "failed to calculate unlocked balance"
+                })?;
+            if available_balance < total_pledge {
+                return Err(actor_error!(
+                    insufficient_funds,
+                    "insufficient funds for aggregate initial pledge requirement {}, available: {}",
+                    total_pledge,
+                    available_balance
+                ));
+            }
+            let needs_cron = !state.deadline_cron_active;
+            state.deadline_cron_active = true;
+
+            state.allocate_sector_numbers(
+                store,
+                &sector_numbers,
+                CollisionPolicy::DenyCollisions,
+            )?;
+
+            state
+                .put_sectors(store, sectors_to_add.clone())
+                .with_context_code(ExitCode::USR_ILLEGAL_STATE, || "failed to put new sectors")?;
+
+            state.assign_sectors_to_deadline(
+                policy,
+                store,
+                rt.curr_epoch(),
+                sectors_to_add,
+                info.window_post_partition_sectors,
+                info.sector_size,
+                params.proving_deadline,
+            )?;
+
+            state
+                .add_initial_pledge(&total_pledge)
+                .with_context_code(ExitCode::USR_ILLEGAL_STATE, || {
+                    "failed to add initial pledgs"
+                })?;
+
+            let fee_to_burn = repay_debts_or_abort(rt, state)?;
+
+            Ok((needs_cron, fee_to_burn))
+        })?;
+
+        burn_funds(rt, fee_to_burn)?;
+
+        let len_for_aggregate_fee = if sectors_len <= 5 { 0 } else { sectors_len - 5 };
```

Comment:
I know the FIP gives no guidance on what to call this magic number but it seems like its something that should live in the policy struct or at the very least in a named constant.

Reply:
> With NI-PoRep, when aggregating at most 5 sectors (i.e. k ≤ 5), the per sector gas cost is already larger than the current value (~34721049), and therefore, in this case, there is no need for any additional fee

I think we could come up with a name

* `ni_aggregate_fee_base_sector_count`
* `base_sector_count_ni_aggregate_fee`

Reply:
aded `NI_AGGREGATE_FEE_BASE_SECTOR_COUNT`